edsnlp/pipelines/misc/dates/dates.py

```diff
@@ -352,85 +149,124 @@ def process(self, doc: Doc) -> List[Span]:
                     self.regex_matcher(
                         sent,
                         as_spans=True,
-                        # return_groupdict=True,
+                        return_groupdict=True,
                     ),
                 )
 
         else:
             dates = self.regex_matcher(
                 doc,
                 as_spans=True,
-                # return_groupdict=True,
+                return_groupdict=True,
             )
 
-        # dates = apply_groupdict(dates)
-
         dates = filter_spans(dates)
-        dates = [date for date in dates if date.label_ != "false_positive"]
+        dates = [date for date in dates if date[0].label_ != "false_positive"]
 
         return dates
 
-    def get_date(self, date: Span) -> Optional[datetime]:
+    def parse(self, dates: List[Tuple[Span, Dict[str, str]]]) -> List[Span]:
         """
-        Get normalised date using `dateparser`.
+        Parse dates using the groupdict returned by the matcher.
 
         Parameters
         ----------
-        date : Span
-            Date span.
+        dates : List[Tuple[Span, Dict[str, str]]]
+            List of tuples containing the spans and groupdict
+            returned by the matcher.
 
         Returns
         -------
-        Optional[datetime]
-            If a date is recognised, returns a Python `datetime` object.
-            Returns `None` otherwise.
+        List[Span]
+            List of processed spans, with the date parsed.
         """
 
-        text_date = date.text
+        for span, groupdict in dates:
+            if span.label_ == "relative":
+                parsed = RelativeDate.parse_obj(groupdict)
+            elif span.label_ == "absolute":
+                parsed = AbsoluteDate.parse_obj(groupdict)
+            else:
+                parsed = Duration.parse_obj(groupdict)
 
-        if date.label_ == "no_day":
-            text_date = "01/" + re.sub(r"[\.\/\s]", "/", text_date)
+            span._.date = parsed
```

Comment:
Why do you use this wording "._."

Reply:
My bad, it comes from spacy extension. Although we mention that it could be interesting to set default ".value" or ".norm" attribute to the BaseComponent

Reply:
That's right, spaCy requires the custom attributes to live in the `Underscore` object for backward-compatibility reasons (see [here](https://spacy.io/usage/processing-pipelines#custom-components-attributes)).

As for the `value`/`norm` extension attributes, I just opened an issue (#47) on the subject, so we can discuss the architecture choices there.

Thanks!